core/src/components/select/select.scss

```diff
@@ -520,7 +535,7 @@ button {
  * is on top of the select. This prevents the label
  * from overlapping any placeholder value.
  */
-:host(.select-label-placement-floating) .native-wrapper .select-placeholder {
+:host(.select-label-placement-floating:not(.label-floating)) .native-wrapper .select-placeholder {
```

Comment:
This block sets `opacity: 0` on the placeholder, and the block below restores the opacity to 1 when the select is expanded, focused, or has a value. However, this PR introduces a fourth case where the label can be floating -- when there's something in the start/end slots. In this case, we want the opacity to remain at the default (neither 0 nor 1). Adding `:not(.label-floating)` here ensures this case is excluded, so the placeholder remains visible even if the select isn't open/focused.

Reply:
Why not update the selectors below to do `:host(select-label-placement-floating.label-floating)`?

Reply:
Because we only want the opacity to be at 1 when the select is expanded, focused, or has a value. Your suggestion would add in the case where there's content in the slots, which would look strange because the select isn't actually being interacted with.

Reply:
My use of the word "restores" might've caused some confusion; the opacity is 0.6 by default, and we want it to stay there unless the label is covering the placeholder (opacity should be 0) or the select is being interacted with (opacity should be 1).

Reply:
Oh that makes sense, thank you! Could we add a comment here as to why we don't use `label-floating`? I can see that tripping folks up. We might also want to update FW-5592 because it sounds like we could then use `.label-floating` if we got rid of the `hasStartEndSlots` condition.

Reply:
Done in https://github.com/ionic-team/ionic-framework/pull/28563/commits/ea9f12bb80e9866982857ea26acaa97529d71528, and updated the tech debt ticket 👍